src/physics/negative_spalart_allmaras_rans_model.cpp

```diff
@@ -0,0 +1,718 @@
+#include <cmath>
+#include <vector>
+#include <complex> // for the jacobian
+
+#include "ADTypes.hpp"
+
+#include "model.h"
+#include "reynolds_averaged_navier_stokes.h"
+#include "negative_spalart_allmaras_rans_model.h"
+
+namespace PHiLiP {
+namespace Physics {
+
+//================================================================
+// Negative Spalart-Allmaras model
+//================================================================
+template <int dim, int nstate, typename real>
+ReynoldsAveragedNavierStokes_SAneg<dim, nstate, real>::ReynoldsAveragedNavierStokes_SAneg(
+    const double                                              ref_length,
+    const double                                              gamma_gas,
+    const double                                              mach_inf,
+    const double                                              angle_of_attack,
+    const double                                              side_slip_angle,
+    const double                                              prandtl_number,
+    const double                                              reynolds_number_inf,
+    const double                                              turbulent_prandtl_number,
+    const double                                              isothermal_wall_temperature,
+    const thermal_boundary_condition_enum                     thermal_boundary_condition_type,
+    std::shared_ptr< ManufacturedSolutionFunction<dim,real> > manufactured_solution_function)
+    : ReynoldsAveragedNavierStokesBase<dim,nstate,real>(ref_length,
+                                                        gamma_gas,
+                                                        mach_inf,
+                                                        angle_of_attack,
+                                                        side_slip_angle,
+                                                        prandtl_number,
+                                                        reynolds_number_inf,
+                                                        turbulent_prandtl_number,
+                                                        isothermal_wall_temperature,
+                                                        thermal_boundary_condition_type,
+                                                        manufactured_solution_function)
+{ }
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_eddy_viscosity (
+    const std::array<real,dim+2> &primitive_soln_rans,
+    const std::array<real,nstate-(dim+2)> &primitive_soln_turbulence_model) const
```

Comment:
same comment as before about creating variables for `dim+2` and `nstate-(dim+2)`